spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -145,6 +146,17 @@ public void setRequestChannelName(String requestChannelName) {
 	}
 
 	/**
+	 * Return the name of the request channel
+	 * @return the channel name.
+	 * @since 4.3
+	 */
+	public String getRequestChannelName() {
+		return this.requestChannel == null ? this.requestChannelName :
+			this.requestChannel instanceof NamedComponent ? ((NamedComponent) this.requestChannel).getComponentName()
+					: null;
```

Comment:
The same for this method.
Also I think we should tweak `MessagingGatewaySupport` a bit and move all those:
```
Assert.state(requestChannel != null,
				"send is not supported, because no request channel has been configured");
```
to the `onInit()` since `requestChannel` is really required.